src/Symfony/Bundle/FrameworkBundle/Tests/Functional/JsonEncoderTest.php

```diff
@@ -44,4 +44,21 @@ public function testDecode()
 
         $this->assertEquals($expected, $decoder->decode('{"@name": "DUMMY", "range": "0..1"}', Type::object(Dummy::class)));
     }
+
+    public function testWarmupEncodableClasses()
+    {
+        static::bootKernel(['test_case' => 'JsonEncoder']);
+
+        // clear already created encoders
+        $encodersDir = \sprintf('%s/json_encoder/encoder/', static::getContainer()->getParameter('kernel.cache_dir'));
+        if (is_dir($encodersDir)) {
+            array_map('unlink', glob($encodersDir.'/*'));
+            rmdir($encodersDir);
+        }
```

Comment:
What about using `Filesystem` for removing the directory and its content?

Also, maybe it can be interesting to remove this directory `setUp` and `tearDown` methods...